### PR TITLE
Fix meta for index_add and index_put

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87775

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: test/test_meta.py [hunk 1]
```diff
@@ -1023,6 +1027,66 @@ def test_fill_alias_relationship(self):
         r2 = torch.ops.aten.fill(inps, 1.0)
         self.assertNotEqual(id(inps), id(r2))
 
+    def get_stride_variants(self, t):
```
Comment (ngimel) [inline test/test_meta.py:1030]: cool! Should we extend all meta tests with inputs generated like this?

**Review by ngimel — APPROVED** [review]

File: torch/_meta_registrations.py [hunk 2]
```diff
@@ -1105,7 +1105,7 @@ def meta_relu_(self):
 
 @register_meta(aten.index_put.default)
 def meta_index_put(self, indices, values, accumulate=False):
-    return self.new_empty(self.size())
+    return torch.empty_like(self)
```
Comment (eellison) [inline torch/_meta_registrations.py:1108]: we should audit the rest of the `new_empty` calls in this file..

File: test/test_meta.py (hunk 1 above)
Comment (SherlockNoMad) [reply test/test_meta.py:1030]: Yes, we can generate strided inputs from existing opinfo test cases.

File: torch/_meta_registrations.py (hunk 2 above)
Comment (SherlockNoMad) [reply torch/_meta_registrations.py:1108]: Yes. We should have instruction on which to use.... 
Stride behavior are subtle for these ops... 
torch.empty
torch.empty_like
torch.new_empty
torch.empty_strided

**Pull request closed without merging**